### DISCUSSION TITLE
i got a problem when i create a function in Openmodelica

### DISCUSSION THREAD

**Discussion by Oh-mymy** [OpenModelica/OpenModelica#9401] [Q&A]
Hello there,this is rookie Mark,recently i got a problem which is about the usage of  function :
when i create a function writen like
"function test1
  input Real x[:];
  output Real y;
algorithm
  y:=x[1];
end test1;
"
Openmodelica will show me an error "[Failed to deduce dimension 1 of x due to missing binding equation.]"
The size of the array i want to input is uncertain, but when i run the same code in Dymola, it works. 
So i wonder how should i do to create a function to input an uncertain size array and output an arbitrary real number?

**Comment by perost** [comment]
The error isn't caused by the function itself but by how you use it. For example, something like this works just fine:
```Modelica
function test1
  input Real x[:];
  output Real y;
algorithm
  y:=x[1];
end test1;

model M
  Real y[:] = {1, 2, 3};
  Real x = test1(y);
end M;
```
So you'll need to show how you actually use the function, otherwise it's hard to say what's wrong.

**Reply by Oh-mymy** [reply]
I find that the error notice could be a little misleading to me, I try your code, it works. Thank you very much for your help, this helped a lot!  

Best,Mark
